### DISCUSSION TITLE
dashboard wont load - grafana timeout

### DISCUSSION THREAD

**Discussion by poccari** [jasonacox/Powerwall-Dashboard#311] [Q&A] [answered]
Hello, 
I noticed a while ago that I couldn't access my grafana instance via my normal browser, so I finally got around to looking into it. 

Looking at the docker logs for the grafana instance I have some errors, so I rebuilt the docker container, and still had issues. I then decided to remove the entire stack (influx, grafana, telegraf etc. ) from docker, re-clone the latest version and re-run the install script to get it fresh, and the error still persisted. Not sure what's going on here, and getting in too deep over my knowledge on how this holds together. 

I'm sure I'm missing something obvious. I have it running on Ubuntu 22.04.1 LTS on an old PC I had lying around. 
Any help/insight would be appreciated. Thanks :) 

Here's the output from running ./verify.sh (I have only pasted the items here where there were issues reported). 
```
Checking grafana
----------------------------------------------------------------------------
 - Config File grafana.env: GOOD
 - Container (grafana): GOOD
 - Service (port 9000): ERROR: Not Listening - Logs:
---
Error: ✗ Get "https://grafana.com/api/plugins/repo/grafana-piechart-panel": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
Error: ✗ Get "https://grafana.com/api/plugins/repo/grafana-piechart-panel": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
Error: ✗ Get "https://grafana.com/api/plugins/repo/grafana-piechart-panel": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
---
 - Version: Grafana CLI version 9.1.2

Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - Service (port 8675): ERROR: Not Listening
 - Version: Unknown
 - Powerwall State: ERROR: Not Connected - Firmware Version: Unknown

```

**Comment by jasonacox** [comment, marked as answer]
> Error: ✗ Get "https://grafana.com/api/plugins/repo/grafana-piechart-panel"

Hi @poccari this means that the Grafana instance can’t reach the internet from you host. Do you have any firewall rules in place that may be restricting this? 

You can install the plugins manually. The dashboard won’t look right until those plugins are installed.

**Comment by poccari** [comment]
Thank you so much! 

Turns out something funky was happening with an update. The firewall rules were all ok, but I thought restarting the docker containers would fix it, but I had to reboot the entire server. 

All working fine now. :) 👍

**Reply by jasonacox** [reply]
Awesome! Thanks for the update.
